### ISSUE TITLE
context should switch in __enter__

### ISSUE THREAD

**Issue by marcellourbani** [ganileni/py_falcon#1]
Hi Orazio
Don't really know python, but this looks like a bug to me.

`        # changes cwd
        os.chdir(self.dir_path)`

should go in __init__

Doesn't affect your use case, but generally speaking a constructor should never have side effects

should one write something like:

```
wd_manager = TempWD(FALCON_DIR)
#wrong directory
print( os.getcwd() )

with wd_manager:
  #right directory
  print( os.getcwd() )

#right directory
print( os.getcwd() )

with wd_manager:
  #wrong directory, again
  print( os.getcwd() )

```

**Comment by ganileni** [comment]
You are perfectly right, thanks! I'll fix it

**Comment by ganileni** [comment]
fixed in a7e187c0140076defee1e89c678ffc96c032c4f8
